src/lib/mina_base/transaction_logic.ml

```diff
@@ -480,7 +480,7 @@ let timing_error_to_user_command_status err =
   | _ ->
       failwith "Unexpected timed account validation error"
 
-let validate_timing_with_min_balance ~account ~txn_amount ~txn_global_slot =
+let validate_timing_with_min_balance' ~account ~txn_amount ~txn_global_slot =
```

Comment:
Can we make this return ```[ `Insufficient_balance of .. | `Invalid_timing of ..]``` instead of `Or_error` and then ignore `Insufficient_balance` in `check_timing` to make sure any new errors here don't break parties?

Reply:
Done. It feels a bit less clean, but I've added a comment that states the invariants; hopefully that should be sufficient.